### PR TITLE
bootupd-validate: remove condition as we have newer bootupd

### LINKED ISSUE #1687: rhel-9.6/c9s: `ext.config.shared.boot.bootupd-validate` failing on ppc64le
```
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 kola-runext-bootupd-validate[2154]: + adopt_and_update
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 kola-runext-bootupd-validate[2154]: + local state_file=/boot/bootupd-state.json
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 kola-runext-bootupd-validate[2154]: + '[' -f /boot/bootupd-state.json ']'
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 kola-runext-bootupd-validate[2154]: + mount -o remount,rw /boot
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 kola-runext-bootupd-validate[2154]: + rm -f /boot/bootupd-state.json
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 kola-runext-bootupd-validate[2169]: + grep 'Adopted and updated.*'
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 kola-runext-bootupd-validate[2168]: + bootupctl adopt-and-update
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 systemd[1]: kola-runext.service: Main process exited, code=exited, status=1/FAILURE
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 systemd[1]: kola-runext.service: Failed with result 'exit-code'.
[2024-12-10T17:31:33.280Z] Dec 10 17:31:27 qemu0 systemd[1]: kola-runext.service: Consumed 11.501s CPU time.
[2024-12-10T17:31:33.280Z] --- FAIL: ext.config.shared.boot.bootupd-validate (56.24s)
```

No interesting additional info in the journal/console that I can see.

@HuijingHei is looking at it.

### PR DESCRIPTION
**Pull request by HuijingHei**
Fixes https://github.com/openshift/os/issues/1687

### CODE REVIEW

**Comment by HuijingHei** [comment]
Hold on, as not yet tested it on ppc64le, will do testing if I could setup the env.

**Comment by HuijingHei** [comment]
Test on ppc64le using `rhcos-419.96.202412161243-0`, result is PASS.
```
[coreos-assembler]$ kola run -E ./test --qemu-image rhcos-419.96.202412161243-0-qemu.ppc64le.qcow2 ext.test.boot.bootupd-validate
=== RUN   ext.test.boot.bootupd-validate
--- PASS: ext.test.boot.bootupd-validate (57.24s)
PASS, output in tmp/kola/qemu-2024-12-17-1255-489
```
Test on ppc64le using `rhcos-419.94.202412142136`, result is PASS.
```
[coreos-assembler]$ kola run -E ./test --qemu-image rhcos-419.94.202412142136-0-qemu.ppc64le.qcow2 ext.test.boot.bootupd-validate
=== RUN   ext.test.boot.bootupd-validate
--- PASS: ext.test.boot.bootupd-validate (43.38s)
PASS, output in tmp/kola/qemu-2024-12-17-1300-559
```
cosa build c9s, and run test, result is PASS.
```
$ kola run -E ./test ext.test.boot.bootupd-validate
=== RUN   ext.test.boot.bootupd-validate                                                                                                                                     
--- PASS: ext.test.boot.bootupd-validate (60.72s) 
```

**Comment by dustymabe** [comment]
considering https://github.com/openshift/os/issues/1687#issuecomment-2548694523 do we still need this?

**Comment by HuijingHei** [comment]
> considering [openshift/os#1687 (comment)](https://github.com/openshift/os/issues/1687#issuecomment-2548694523) do we still need this?

Thanks for the pointer, refresh the code to remove the condition as we have the newer bootupd (https://issues.redhat.com/browse/RHEL-71040), could you help to review, thanks!

File: tests/kola/boot/bootupd-validate [hunk 1]
```diff
@@ -40,7 +38,7 @@ overwrite_file() {
             overwrite=1
             ;;
         *)
-            echo "skipped overwrite"
+            echo "skipped overwrite on none esp device"
```
Comment (dustymabe) [inline tests/kola/boot/bootupd-validate]: hmm. not sure what we are saying here.

maybe `"skipped overwrite on platform without esp device"` would be better?

File: tests/kola/boot/bootupd-validate [hunk 2]
```diff
@@ -81,7 +77,7 @@ validate() {
     esac
 }
 
-# should remove this until https://issues.redhat.com/browse/RHEL-66584 is fixed
+# should remove this until https://issues.redhat.com/browse/RHEL-70849 is fixed
```
Comment (dustymabe) [inline tests/kola/boot/bootupd-validate]: maybe "should remove this when" instead of `until`?

**Review by dustymabe — DISMISSED** [review]
LGTM but suggested some better wording

**Review by dustymabe — APPROVED** [review]

**Pull request merged**